src/vs/workbench/parts/execution/electron-browser/terminalService.ts

```diff
@@ -49,14 +72,41 @@ export class MacTerminalService implements ITerminalService {
 			child.on('exit', (code: number) => {
 				c(code === 0 ? 'iterm.scpt' : 'terminal.scpt');
 			});
-		}).then(name => uri.parse(require.toUrl(`vs/workbench/parts/execution/electron-browser/${ name }`)).fsPath);
+		}).then(name => uri.parse(require.toUrl(`vs/workbench/parts/execution/electron-browser/${name}`)).fsPath);
 	}
 }
 
 export class LinuxTerminalService implements ITerminalService {
 	public serviceId = ITerminalService;
 
+	constructor(
+		@IConfigurationService private _configurationService: IConfigurationService
+	) { }
+
+
 	public openTerminal(path: string): void {
-		cp.spawn('x-terminal-emulator', [], { cwd: path });
+		this._configurationService.loadConfiguration().done(configuration => {
+			return new Promise((success, failed) => {
+				this.spawnTerminal(
+					cp,
+					configuration,
+					path,
+					success,
+					err => {
+						errors.onUnexpectedError(err);
+						failed(err);
+					}
+				);
+			});
+		}, errors.onUnexpectedError);
 	}
+
+	private spawnTerminal(spawner, configuration, path: string, onExit, onError) {
+		let terminalConfig = configuration.terminal;
+		let exec = terminalConfig.linux.exec || defaultLinuxTerm;
+		const child = spawner.spawn(exec, [], { cwd: path });
```

Comment:
node runs `gnome-terminal` at cwd specified in the options. I tested on Fedora 23 and it does go to the cwd specified. I think it's just a useful flag when someone wants to run from a different path other than the cwd `gnome-terminal` was executed from.